library/ssl_tls.c

```diff
@@ -808,6 +849,9 @@ int mbedtls_ssl_derive_keys( mbedtls_ssl_context *ssl )
         handshake->tls_prf = ssl3_prf;
         handshake->calc_verify = ssl_calc_verify_ssl;
         handshake->calc_finished = ssl_calc_finished_ssl;
+#if defined(MBEDTLS_SSL_EXPORT_KEYS)
+        tls_prf_type = MBEDTLS_SSL_TLS_PRF_SSL3;
```

Comment:
I think it would be safer if we would set the `tls_prf_type` based on `handshake->tls_prf` and would do so right before we use it. For the sake of readability we could add a static function `tls_prf_get_type()` that receives a function pointer and returns an enum value of the appropriate type.

Reply:
I think it's safer \ simpler to set this variable when we know what the `tls_prf` function type is, hence here.
I think it's less safer to add this logic, for getting the type, by comparing the function pointer with the static functions that are or aren't defined, according to precompilation configuration

Reply:
I agree that it is simpler, but I think that it is not safer. If we set them separately, we need to make sure that the values of  `tls_prf_type` and `handshake->tls_prf` are synchronised at all times. This is error prone and harder to maintain.

If we set `tls_prf_type` based on the`handshake->tls_prf`  function pointer, then as you say, we need to take into account the compile time options too. This makes the implementation slightly more complicated. I suggested to move this translation into the static function  `tls_prf_get_type()` to mitigate the complexity problem and preserve readability.